src/metabase/driver/crate/query_processor.clj

```diff
@@ -0,0 +1,18 @@
+(ns metabase.driver.crate.query-processor
+  (:require [korma.core :as k]
+            [metabase.driver.generic-sql.query-processor :as qp])
+  (:import (metabase.query_processor.interface ComparisonFilter CompoundFilter)))
+
+
+(defn- rewrite-between
+  "Rewrite [:between <field> <min> <max>] -> [:and [:>= <field> <min>] [:<= <field> <max>]]"
+  [clause]
+  (CompoundFilter. :and [(ComparisonFilter. :>= (:field clause) (:min-val clause))
+                         (ComparisonFilter. :<= (:field clause) (:max-val clause))]))
+
+(defn apply-filter
+  "Apply custom generic SQL filter. This is the place to perform query rewrites."
+  [_ korma-form {clause :filter}]
+  (k/where korma-form (case (:filter-type clause)
+            :between (qp/filter-clause->predicate (rewrite-between clause))
```

Comment:
since Crate does not support the `BETWEEN` predicate, we had to rewrite it to `<min> <= <value> <= <max>`
not sure if this is the correct place, it was definitely the easiest, though ;)